### PR TITLE
New ADR - message-broker

### PR DESCRIPTION
**Pull request by anna-knudsen**

### CODE REVIEW

**Review by Exobitt — APPROVED** [review]

File: doc/architecture/adr/0019-message-broker.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+ActiveMQ:
+Active MQ is mostly for Java.
+
+Redis:
+Redis doesn't have built-in SSL and offers no persistence out of the box.
+
+### Positive Consequences
+
+Easy setup
+Lots of documentation
```
Comment (duizer) [inline doc/architecture/adr/0019-message-broker.md:48]:
```suggestion
* Easy setup
* Lots of documentation
```

File: doc/architecture/adr/0019-message-broker.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+---
+
+## Decision Outcome
+
+We are going with RabbitMQ
+
+## Rationale
+
+RabbitMQ seems to be a simple message broker and not much else, which is exactly what we need.
```
Comment (duizer) [inline doc/architecture/adr/0019-message-broker.md:31]: This is a minor thing: The other paragraphs starts with the name of the technology and then a colon. This section does not.

File: doc/architecture/adr/0019-message-broker.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+It is a very popular, open source message broker with lots of support and documentation.
+It also seems very easy to set up.
+
+Kafka:
```
Comment (duizer) [inline doc/architecture/adr/0019-message-broker.md:35]: The rendered page will not have a break line. So maybe:

* Make this bold
* Use a header (###)
* Delete the line

File: doc/architecture/adr/0019-message-broker.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
+Message broker
+
+* Status:
```
Comment (duizer) [inline doc/architecture/adr/0019-message-broker.md:3]: Please set Status to "Proposed". Also add "PROPOSAL" to the file name as described here: https://github.com/Energinet-DataHub/energy-origin/blob/main/doc/architecture/adr/process.md. The process description is unfortunately not 100% updated, so please also update the README file with a link to this ADR.

File: doc/architecture/adr/0019-message-broker.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+* Deciders: @anna-knudsen, @duizer, @ckr123, @Exobitt
+* Date: 2022-01-19
+
+---
+
+## Context and Problem Statement
+
+For our event driven architecture solution we need a message broker that can handle a large number of messages.
+We also want a message broker that is fairly easy to set up.
```
Comment (duizer) [inline doc/architecture/adr/0019-message-broker.md:12]: I suggest that the Context states that this is for the Certificates domain specifically. ADR's are for the entire system, but in this case it is based on the needs within the Certificates domain.

File: doc/architecture/adr/0019-message-broker.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+Redis doesn't have built-in SSL and offers no persistence out of the box.
+
+### Positive Consequences
+
+Easy setup
+Lots of documentation
+
+### Negative Consequences
+
+We are not yet 100% sure that RabbitMQ can handle the number of messages we need, and we are aware that RabbitMQ can not be scaled to the extent that Kafka can.
```
Comment (duizer) [inline doc/architecture/adr/0019-message-broker.md:52]: Is it possible to add the number of messages that we currently think that we need?

**Review by duizer — CHANGES_REQUESTED** [review]

**Review by duizer — APPROVED** [review]

**Review by Exobitt — APPROVED** [review]
lgtm

**Pull request merged**